src/materials/SteelCreepDamage.C

```diff
@@ -0,0 +1,234 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SteelCreepDamage.h"
+
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+#include "ElasticityTensorTools.h"
+#include "MooseUtils.h"
+#include <cmath>
+#include "RankTwoScalarTools.h"
+
+#include "libmesh/utility.h"
+
+registerMooseObject("BlackBearApp", SteelCreepDamage);
+registerMooseObject("BlackBearApp", ADSteelCreepDamage);
+
+template <bool is_ad>
+InputParameters
+SteelCreepDamageTempl<is_ad>::validParams()
+{
+  InputParameters params = ScalarDamageBaseTempl<is_ad>::validParams();
+  params.addClassDescription("Steel scalar damage model: Material 'suddenly' loses load-carrying "
+                             "capacity. This can model, e.g., 316H creep failure. See "
+                             "'Creep failure simulations of 316H at 550C: Part I - A method and "
+                             "validation, by Oh et al, Engineering Fracture Mechanics 78 (2011)'");
+  params.addRequiredParam<Real>(
+      "epsilon_f",
+      "epsilon_f parameter refers to uniaxial creep fracture strain (creep ductility).");
+  params.addRequiredParam<Real>("creep_law_exponent", "Exponent of the creep power law.");
+  params.addParam<Real>(
+      "reduce_factor",
+      1000.0,
+      "reduce_factor parameter refers to a reduction on the load-carrying capacity (stress).");
+  params.addRangeCheckedParam<Real>("initial_damage_value",
```

Comment:
This parameter name is potentially a little misleading. It makes it sound like you are starting out with a damaged material. Maybe rename it to 'reduction_damage_threshold' or something like that?

Reply:
`reduction_damage_threshold` sounds as good to me as it can with three words.